### PR TITLE
receiver/prometheus: add metricGroup.toSummaryPoint pdata conversion

### PR DESCRIPTION
**Pull request by odeke-em**
Implements metricGroupPdata toSummaryPoint and added unit tests
as well as equivalence tests to ensure the migration will render
the same results.
    
Updates #3137
Depends on PR #3667
Updates PR #3427

### CODE REVIEW

**Comment by odeke-em** [comment]
Another one, kindly cc-ing @bogdandrutu @Aneurysm9 @rakyll @alolita @tigrannajaryan @anuraaga

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go [hunk 1]
```diff
@@ -96,6 +96,89 @@ func (mf *metricFamilyPdata) updateLabelKeys(ls labels.Labels) {
...
+
+func (mg *metricGroupPdata) toDistributionPoint(orderedLabelKeys []string, dest *pdata.HistogramDataPointSlice) bool {
+	if !mg.hasCount || len(mg.complexValue) == 0 {
+		return false
+	}
+
+	mg.sortPoints()
+
+	// for OCAgent Proto, the bounds won't include +inf
+	// TODO: (@odeke-em) should we also check OpenTelemetry Pdata for bucket bounds?
```
Comment (bogdandrutu) [inline receiver/prometheusreceiver/internal/otlp_metricfamily.go:112]: https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L414

**Review by dashpole — APPROVED** [review]
toSummaryPoint looks good.

**Comment by bogdandrutu** [comment]
@dashpole indeed the Summary looks good but depends on a PR with some comments :)

**Comment by dashpole** [comment]
Yes.  This is not ready to merge.

**Review by Aneurysm9 — APPROVED** [review]

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go (hunk 1 above)
Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:112]: I filed https://github.com/open-telemetry/opentelemetry-collector/issues/3684 as this change is a translation.

**Review by cuonglm — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by odeke-em** [comment]
> Please rebase

Rebase done, thanks @bogdandrutu!

**Pull request merged**